src/main/java/de/tum/cit/aet/artemis/core/web/FileResource.java

```diff
@@ -587,6 +592,66 @@ public ResponseEntity<byte[]> getAttachmentUnitAttachmentSlide(@PathVariable Lon
         }
     }
 
+    /**
+     * GET files/slides/{slideId} : Get the lecture unit attachment slide by slide id
+     *
+     * @param slideId the id of the slide that wanted to be retrieved
+     * @return The requested file, 403 if the logged-in user is not allowed to access it, or 404 if the file doesn't exist
+     */
+    @GetMapping("files/slides/{slideId}")
+    @EnforceAtLeastStudent
+    public ResponseEntity<byte[]> getSlideById(@PathVariable Long slideId) {
+        log.debug("REST request to get the slide : {}", slideId);
+
+        Slide slide = slideRepository.findById(slideId).orElseThrow(() -> new EntityNotFoundException("Slide", slideId));
+
+        AttachmentUnit attachmentUnit = slide.getAttachmentUnit();
+
+        if (slide.getHidden() != null) {
+            throw new AccessForbiddenException("Slide is hidden");
+        }
+
+        String directoryPath = slide.getSlideImagePath();
+
+        // Use regular expression to match and extract the file name with ".png" format
+        Pattern pattern = Pattern.compile(".*/([^/]+\\.png)$");
+        Matcher matcher = pattern.matcher(directoryPath);
+
+        if (matcher.matches()) {
+            String fileName = matcher.group(1);
+            return buildFileResponse(
```

Comment:
maybe we can extend `getActualPathFromPublicPathString(..)` to also handle these cases here, just to be more consistent?